### PR TITLE
Bump default API version to V36.0 (VCD 10.3+)

### PR DESCRIPTION
**Pull request by Didainius**
VCD 10.2 is going EOL for 2022-10-15 and the next release is going to be later than that therefore bumping default API version to V36.0. (for those with explicit needs there is an option to override the default version by using `GOVCD_API_VERSION` environment variable).

Additionally, this PR removes conditional checks that were being used for versions < 36.0

It should also fix all currently known flaky tests, which should make test run clean:
* Auth endpoint returned URLs changed

From
```xml
<VersionInfo deprecated="true">
    <Version>35.0</Version>
    <LoginUrl>https://HOST/api/sessions</LoginUrl>
</VersionInfo>

```

To
```xml
<VersionInfo deprecated="false">
  <Version>36.0</Version>
  <LoginUrl>https://HOST/cloudapi/1.0.0/sessions</LoginUrl>
  <ProviderLoginUrl>https://HOST/cloudapi/1.0.0/sessions/provider</ProviderLoginUrl>
</VersionInfo>

```

Test `TestClient_getloginurl` started failing immediately because it relied on a matching path which has changed.

* VCD introduced new state for VM `PARTIALLY_POWERED_OFF` in 10.3. Test `Test_VMToggleHardwareVirtualization` failed because function `ToggleHardwareVirtualization` specifically relied on VM being `POWERED_OFF`. This function is adjusted to also match `PARTIALLY_POWERED_OFF`
```
START: vm_test.go:610: TestVCD.Test_VMToggleHardwareVirtualization
vm_test.go:646:
    check.Assert(err, IsNil)
... value *errors.errorString = &errors.errorString{s:"hardware virtualization can be changed from powered off state, status: PARTIALLY_POWERED_OFF"} ("hardware virtualization can be changed from powered off state, status: PARTIALLY_POWERED_OFF")

FAIL: vm_test.go:610: TestVCD.Test_VMToggleHardwareVirtualization
```

* VCD introduced new state for VM `PARTIALLY_POWERED_OFF` in 10.3. `Test_VMPowerOnPowerOff` started failing.
```
START: vm_test.go:666: TestVCD.Test_VMPowerOnPowerOff
vm_test.go:710:
    check.Assert(vmStatus, Equals, "POWERED_OFF")
... obtained string = "PARTIALLY_POWERED_OFF"
... expected string = "POWERED_OFF"

FAIL: vm_test.go:666: TestVCD.Test_VMPowerOnPowerOff
```

* Fix `Test_RemoveAllNetworks` by ensureing vApp is powered off before removing all networks.
```
START: vapp_test.go:611: TestVCD.Test_RemoveAllNetworks
vapp_test.go:670:
    check.Assert(err, IsNil)
... value *errors.errorString = &errors.errorString{s:"task did not complete successfully:  [409:INVALID_STATE] - [ 35010cb4-48ac-4fe2
-a4c2-f6af984e243c ] The requested operation could not be executed on vApp \"f54723e7-638f-4255-8ca0-0255c9c93e29\". Stop the vApp and
 try again."} ("task did not complete successfully:  [409:INVALID_STATE] - [ 35010cb4-48ac-4fe2-a4c2-f6af984e243c ] The requested oper
ation could not be executed on vApp \"f54723e7-638f-4255-8ca0-0255c9c93e29\". Stop the vApp and try again.")

FAIL: vapp_test.go:611: TestVCD.Test_RemoveAllNetworks
```

### CODE REVIEW

**Review by lvirbalas — APPROVED** [review]
This is great, thank you! As I will be away for a few days, I'm giving an early LGTM, so you don't have to wait on me for merging.

My only request is to see what the whole test suite returns after this change, hopefully nothing complains :)

**Comment by Didainius** [comment]
> My only request is to see what the whole test suite returns after this change, hopefully, nothing complains :)

Yeah. I am now fixing things up, but tests will take their time before this lands.

File: govcd/common_test.go [hunk 1]
```diff
@@ -230,10 +230,10 @@ func testCheckLoadBalancerConfig(beforeLb *types.LbGeneralParamsWithXml, beforeL
 // reached
 func isTcpPortOpen(host, port string, timeout int) bool {
 	retryTimeout := timeout
-	// due to the VMs taking long time to boot it needs to be at least 5 minutes
+	// due to the VMs taking long time to boot it needs to be at least 6 minutes
 	// may be even more in slower environments
-	if timeout < 5*60 { // 5 minutes
-		retryTimeout = 5 * 60 // 5 minutes
+	if timeout < 6*60 { // 5 minutes
```
Comment (adambarreiro) [inline govcd/common_test.go:235]:
```suggestion
	if timeout < 6*60 { // 6 minutes
```
or we can get rid of these comments as the same info is on line 233 as well.

**Review by adambarreiro — APPROVED** [review]
Nice 😄

File: govcd/common_test.go (hunk 1 above)
Comment (Didainius) [reply govcd/common_test.go]: Removed it in fact. Even the syntax is quite clear `6 * 60` + there is one explanation comment.

**Pull request merged**